### PR TITLE
Detect failure in any ran suite, so that the coordinator can throw

### PR DESCRIPTION
**Pull request by edmundnoble**
Not 100% sure about this approach, but it does let me throw exceptions in the coordinating code to signify failure, which is sufficiently noisy and what CI expects.

### CODE REVIEW

**Comment by edmundnoble** [comment]
Hold on, I have to update the docs.

File: core/src/main/scala/testz/data.scala [hunk 1]
```diff
@@ -93,3 +93,24 @@ object Result {
...
+)
+
+object TestOutput {
+  def combine(fst: TestOutput, snd: TestOutput) =
+    new TestOutput(
+      fst.failed || snd.failed,
+      { () => fst.print(); snd.print() }
+    )
+
+  def combineAll1(output1: TestOutput, outputs: TestOutput*) = {
```
Comment (hrhino) [inline core/src/main/scala/testz/data.scala:109]: arguably this is monoidal and not just semigroupal, but I'm sure you considered that.

File: docs/src/main/tut/docs/05-specs2.md [hunk 2]
```diff
@@ -5,4 +5,4 @@ title: specs2
 
 # {{ page.title }}
 
-testz provides a module for specs2 compatibility.
\ No newline at end of file
+testz will soon provide a module for specs2 compatibility.
```
Comment (hrhino) [inline docs/src/main/tut/docs/05-specs2.md:8]: aw...

File: runner/src/main/scala/testz/runner/Runner.scala [hunk 3]
```diff
@@ -104,10 +106,10 @@ object Runner {
    * It takes a list of `() => () => Future[Unit]` and runs all of them in
```
Comment (hrhino) [inline runner/src/main/scala/testz/runner/Runner.scala]: newly inaccurate doc comment

File: util/src/main/scala/testz/futureUtil.scala [hunk 4]
```diff
@@ -34,6 +34,26 @@ object futureUtil {
...
+          if (ne.isCompleted) {
+            inner(acc || ne.value.get.get)
+          } else {
+            ne.flatMap(b => outer(acc || b))(ec)
+          }
+        } else {
+          Future.successful(acc)
+        }
+      }
+      inner(acc)
```
Comment (hrhino) [inline util/src/main/scala/testz/futureUtil.scala:52]: oh I get it. this makes `inner` tail recursive in the `ne.isCompleted` branch.

File: util/src/main/scala/testz/futureUtil.scala [hunk 5]
```diff
@@ -52,25 +72,25 @@ object futureUtil {
     inner()
   }
 
-  def collectIterator[A](it: Iterator[Future[() => Unit]])(ec: ExecutionContext): Future[() => Unit] = {
+  def collectIterator[A](it: Iterator[Future[A]])(ec: ExecutionContext): Future[List[A]] = {
```
Comment (hrhino) [inline util/src/main/scala/testz/futureUtil.scala]: doesn't this reverse the order of elements gotten from `it`? could that make a difference to anyone?

File: tests/src/main/scala/testz/Main.scala [hunk 6]
```diff
@@ -45,15 +45,16 @@ object Main {
...
-    val suites: List[() => Future[() => Unit]] = List(
+    val suites: List[() => Future[TestOutput]] = List(
       () => runPure("Stdlib Unit Tests", (new StdlibSuite).tests(harness)),
       () => runPure("Exhaustive tests", (new ExhaustiveSuite).tests(harness))
     )
 
-    Await.result(Runner(suites, global), Duration.Inf)
+    val result = Await.result(Runner(suites, global), Duration.Inf)
+
+    if (result.failed) throw new Exception("some tests failed")
```
Comment (hrhino) [inline tests/src/main/scala/testz/Main.scala:58]: where do the results get printed?

File: docs/src/main/tut/docs/05-specs2.md (hunk 2 above)
Comment (edmundnoble) [reply docs/src/main/tut/docs/05-specs2.md:8]: It's #14 ;)

File: runner/src/main/scala/testz/runner/Runner.scala (hunk 3 above)
Comment (edmundnoble) [reply runner/src/main/scala/testz/runner/Runner.scala]: Yep, will fix asap.

File: util/src/main/scala/testz/futureUtil.scala (hunk 4 above)
Comment (edmundnoble) [reply util/src/main/scala/testz/futureUtil.scala:52]: Right, this code is fucking abhorrent but this is what I need to do for fast code inside `Future`. 🗡

File: tests/src/main/scala/testz/Main.scala (hunk 6 above)
Comment (edmundnoble) [reply tests/src/main/scala/testz/Main.scala:58]: Ask your `PureHarness`; you told it how to print them when you made it.

File: util/src/main/scala/testz/futureUtil.scala (hunk 5 above)
Comment (edmundnoble) [reply util/src/main/scala/testz/futureUtil.scala]: Hmmm probably. Good to know, deserves a comment.

File: core/src/main/scala/testz/data.scala (hunk 1 above)
Comment (edmundnoble) [reply core/src/main/scala/testz/data.scala:109]: Exactly, people shouldn't have empty test suites.

Comment (hrhino) [reply core/src/main/scala/testz/data.scala:109]: but I can do it with scalatest!

File: util/src/main/scala/testz/futureUtil.scala (hunk 5 above)
Comment (edmundnoble) [reply util/src/main/scala/testz/futureUtil.scala]: Actually, the synchronous elements were prepended, and the async ones were appended!

Fixed this and made it not reversed.

**Review by hrhino — COMMENTED** [review]
lgtm now

**Review by hrhino — APPROVED** [review]

**Pull request merged**